java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -37,62 +38,97 @@
 public class RangeEqualsVisitor {
 
   protected final ValueVector right;
-  protected final int leftStart;
-  protected final int rightStart;
-  protected final int length;
+  protected int leftStart;
+  protected int rightStart;
+  protected int length;
+
+  protected boolean typeCheckNeeded = true;
+
+  public void setTypeCheckNeeded(boolean typeCheckNeeded) {
```

Comment:
If you have this flag in the constructor, then do you need this API also?

Reply:
You are right, removed.